rust/src/dcerpc/dcerpc.rs

```diff
@@ -510,6 +524,62 @@ impl DCERPCState {
         self.prev_tx_call_id = call_id;
     }
 
+    pub fn parse_data_gap(&mut self, direction: u8) -> AppLayerResult {
+        match direction {
+            core::STREAM_TOSERVER => {
+                self.ts_gap = true;
+                self.ts_ssn_gap = true;
+            },
+            _ => {
+                self.tc_gap = true;
+                self.tc_ssn_gap = true;
+            },
+        }
+        AppLayerResult::ok()
+    }
+
+    pub fn post_gap_housekeeping(&mut self, dir: u8) {
+        SCLogDebug!("ts ssn gap: {:?}, tc ssn gap: {:?}, dir: {:?}", self.ts_ssn_gap, self.tc_ssn_gap, dir);
+        if self.ts_ssn_gap && dir == core::STREAM_TOSERVER {
+            for tx in &mut self.transactions {
+                if tx.id >= self.tx_id {
+                    SCLogDebug!("post_gap_housekeeping: done");
+                    break;
+                }
+                if tx.req_done == false {
+                    tx.req_lost = true;
+                }
+                tx.req_done = true;
+            }
+        } else if self.tc_ssn_gap && dir == core::STREAM_TOCLIENT {
+            for tx in &mut self.transactions {
+                if tx.id >= self.tx_id {
+                    SCLogDebug!("post_gap_housekeeping: done");
+                    break;
+                }
+                if tx.req_done == false {
+                    tx.req_lost = true;
+                }
+                if tx.resp_done == false {
+                    tx.resp_lost = true;
+                }
+                tx.req_done = true;
+                tx.resp_done = true;
+            }
+        }
+    }
+
+    pub fn search_dcerpc_record<'a>(&mut self, i: &'a[u8]) -> nom::IResult<&'a[u8], &'a[u8]> {
+        let mut d = i;
+        while d.len() >= 2 {
```

Comment:
Why change `d` and not have an index variable ?

Reply:
I suppose that could be done. Does that lead to improved handling and/or performance?

Reply:
I guess.
The classical way is more using an index...